component/attr_cache/attr_cache.go

```diff
@@ -164,50 +163,56 @@ func (ac *AttrCache) OnConfigChange() {
 }
 
 // Helper Methods
-// deleteDirectory: Marks a directory and all its contents deleted.
-// This should only be called when ac.cacheDirs is false.
-// This marks items deleted instead of invalidating them.
-// That way if a request came in for a deleted item, we can respond from the cache.
-func (ac *AttrCache) deleteDirectory(path string, time time.Time) {
-	//get attrCacheItem
-	toBeDeleted, getErr := ac.cacheMap.get(path)
-	if getErr != nil || !toBeDeleted.exists() {
+
+// Record that the directory and all its contents have been deleted from the file system.
+// This should only be called when cacheDirs is false.
+// If cacheDirs is true, use deleteCachedDirectory instead.
+func (ac *AttrCache) deleteDirectory(path string, deletedAt time.Time) {
+	// get the entry to be marked deleted
+	value, found := ac.cache.get(path)
+	if !found {
+		// the directory is not in the cache
+		log.Warn("AttrCache::deleteDirectory : %s no cache record of directory", path)
+		// record the fact that the directory has been deleted
+		ac.cache.insert(internal.CreateObjAttrDir(path), false, deletedAt)
+	} else if !value.exists() {
+		// the directory has already been marked deleted
+		// cacheDirs=false, so attr_cache is not responsible for returning ENOENT
+		// log a warning but do nothing
 		log.Warn("AttrCache::deleteDirectory : %s directory does not exist", path)
-		return
+	} else {
+		// record that the entry and all its children have been deleted
+		value.markDeleted(deletedAt)
 	}
-	toBeDeleted.markDeleted(time)
 }
 
-// deleteCachedDirectory: marks a directory and all its contents deleted
-// while keeping directory cache coherent.
-// This should only be called when ac.cacheDirs is true.
-// This marks items deleted instead of invalidating their entries.
-// That way if a request comes in for a deleted item, it's still a cache hit.
+// Record that the directory and all its contents have been deleted from the file system.
+// This should only be called when cacheDirs is true.
 func (ac *AttrCache) deleteCachedDirectory(path string, time time.Time) error {
-
-	// Delete all descendants of the path, then delete the path
-
-	//get attrCacheItem
-	toBeDeleted, getErr := ac.cacheMap.get(path)
-	if getErr != nil || !toBeDeleted.exists() {
-		log.Err("AttrCache::deleteCachedDirectory : %s not found", path)
+	// get the item
+	item, found := ac.cache.get(path)
+	if !found || !item.exists() {
```

Comment:
I see you are using the ```||``` comparison here but you have previously split this up into an ```else if``` in deleteDirectory(). Is this trivial? is it for more readability? I'm just noticing the style of error checking is not too consistent so far

Reply:
In `deleteDirectory`, I chose to take two different actions for the two cases:
- if there is no entry at all (`!found`), I add a new entry which is marked deleted
- if there is an entry which is marked deleted (`!value.exists()`), I do nothing (but log a warning)
But here I do the same thing for both cases (log and return an error).